Source/Serilog.Exceptions/Destructurers/ReflectionBasedDestructurer.cs

```diff
@@ -95,12 +95,13 @@ public ReflectionBasedDestructurer(int destructuringDepth)
             AppendTypeIfPossible(propertiesBag, exception.GetType());
         }
 
-        private static string GetOrGenerateRefId(ref int nextCyclicRefId, IDictionary<string, object> destructuredObject)
+        private static string? GetOrGenerateRefId(ref int nextCyclicRefId, IDictionary<string, object?> destructuredObject)
         {
-            string refId;
+            string? refId;
             if (destructuredObject.ContainsKey(IdLabel))
             {
-                refId = (string)destructuredObject[IdLabel];
+                var objectValue = destructuredObject[IdLabel];
+                refId = (string?)objectValue;

```

Comment:
I guess it is just necessary because `refId` is a `string` and `destructuredObject` dictionary contains `object` values.
For this particular special key `IdLabel` we know for sure that it contains a string because we put it there ourselves.

Reply:
Is there a better way we can handle that?

Reply:
I guess a natural way to improve (at least in one dimension) would be to introduce a layer of abstraction and replace `object` values with some explicit domain representation of `DestructuredValue` and `ReferenceId`. 

Other than that, my call is that such "low level" code is well placed in an internal class (instead of public API). Of course, I am open to suggestions.

Reply:
I think that would make a good subsequent PR,